File: apps/desktop/src/api/llm-settings.ts
import { apiGet, apiPut } from "./client"

export type LlmProviderId = "deepseek" | "openai_compatible"

export interface LlmProviderOption {
  id: LlmProviderId
  label: string
  requires_base_url: boolean
  default_model: string
  default_base_url: string
}

export interface LlmSettings {
  provider: LlmProviderId
  model: string
  base_url: string
  providers: LlmProviderOption[]
}

export interface LlmSettingsUpdate {
  provider: LlmProviderId
  model: string
  base_url: string
}

const LLM_SETTINGS_PATH = "/api/settings/llm"

export function getLlmSettings(): Promise<LlmSettings> {
  return apiGet<LlmSettings>(LLM_SETTINGS_PATH)
}

export function updateLlmSettings(payload: LlmSettingsUpdate): Promise<LlmSettings> {
  return apiPut<LlmSettings, LlmSettingsUpdate>(LLM_SETTINGS_PATH, payload)
}

File: backend/models/llm_settings.py
"""Safe request and response contracts for local LLM configuration."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

AIProviderName = Literal["deepseek", "openai_compatible"]
class LLMProviderOption(BaseModel):
    id: AIProviderName
    label: str
    requires_base_url: bool
    default_model: str
    default_base_url: str


class LLMSettingsResponse(BaseModel):
    provider: AIProviderName
    model: str
    base_url: str
    providers: list[LLMProviderOption] = Field(default_factory=list)


class LLMSettingsUpdateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    provider: AIProviderName
    model: str = Field(min_length=1, max_length=256)
    base_url: str = Field(default="", max_length=2048)


__all__ = [
    "AIProviderName",
    "LLMProviderOption",
    "LLMSettingsResponse",
    "LLMSettingsUpdateRequest",
]
